Adds StorageConsumer, supporting all storage except Elasticsearch AWS

codefromthecrypt opened the pull request:
This consumer writes spans to a Zipkin `StorageComponent`, such as
Elasticsearch.

In order to connect, you minimally need to set `zipkin.storage.type` to
a configured backend. You also need to set relevant properties for that.

Ex.
```bash
java -jar zipkin-sparkstreaming-job.jar \
  --zipkin.storage.type=elasticsearch \
  --zipkin.storage.elasticsearch.hosts=http://127.0.0.1:9200 \
  ...
```

Configuration properties can be set via commandline parameters, system
properties or any other alternative [supported by Spring Boot](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html).

Besides setting storage properties, you can also override the following.
All of the below have a prefix of "zipkin.sparkstreaming.consumer.storage"

Property | Default |Description
--- | --- | ---
fail-fast | true | check storage before submitting the job.

Fixes #8

adriancole commented on pom.xml:
```diff
@@ -47,7 +47,7 @@
     <spark.version>1.6.3</spark.version>
 
-    <zipkin.version>1.20.0</zipkin.version>
+    <zipkin.version>1.20.1-SNAPSHOT</zipkin.version>
```
pending release of code that makes storage property types serializable

adriancole commented on sparkstreaming/src/test/java/zipkin/sparkstreaming/StorageConsumerTest.java:
```diff
@@ -0,0 +1,24 @@
+
+  @Test
+  public void test() {
+    // TODO
```
I will write this.. currently tested manually both in dev mode and also to a spark cluster

adriancole commented:
This now supports all storage except Elasticsearch AWS

adriancole commented:
will rebase this on zipkin 1.20.1 once syncs to maven central

naoman reviewed (approved):
Looks good to me!

Commit:
Adds StorageConsumer

This consumer writes spans to a Zipkin `StorageComponent`, such as
Elasticsearch.

In order to connect, you minimally need to set `zipkin.storage.type` to
a configured backend. You also need to set relevant properties for that.

Ex.
```bash
java -jar zipkin-sparkstreaming-job.jar \
  --zipkin.storage.type=elasticsearch \
  --zipkin.storage.elasticsearch.hosts=http://127.0.0.1:9200 \
  ...
```

Configuration properties can be set via commandline parameters, system
properties or any other alternative [supported by Spring Boot](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html).

Besides setting storage properties, you can also override the following.
All of the below have a prefix of "zipkin.sparkstreaming.consumer.storage"

Property | Default |Description
--- | --- | ---
fail-fast | true | check storage before submitting the job.

adriancole commented:
ok backfilled tests as promised.
